processor/internal/transformer/destination_transformer/embedded/pubsub/pubsub_test.go

```diff
@@ -0,0 +1,676 @@
+								"attr-key-default": "value-default",
+							},
+							"topicId": "topic-default",
+							"userId":  "",
+						},
+						StatusCode: http.StatusOK,
+						Metadata:   types.Metadata{},
+					},
+					{
+						Output: map[string]interface{}{
+							"message": map[string]interface{}{
+								"type":       "identify",
+								"attr-key-1": "value-1",
+								"attr-key-3": "value-3",
+							},
+							"attributes": map[string]interface{}{
+								"attr-key-1": "value-1",
+								"attr-key-3": "value-3",
+							},
+							"topicId": "topic-default",
+							"userId":  "",
+						},
+						StatusCode: http.StatusOK,
+						Metadata:   types.Metadata{},
+					},
+					{
+						Output: map[string]interface{}{
+							"message": map[string]interface{}{
+								"type":       "track",
+								"event":      "event-A",
+								"attr-key-1": "value-1",
+								"properties": map[string]interface{}{
+									"attr-key-2": "value-2",
+									"A": map[string]interface{}{
+										"B": map[string]interface{}{
+											"attr-key-6": "value-6",
+										},
+									},
+								},
+								"context": map[string]interface{}{
+									"traits": map[string]interface{}{
+										"attr-key-3": "value-3",
+										"A": map[string]interface{}{
+											"B": map[string]interface{}{
+												"attr-key-4": "value-4",
+												"C": map[string]interface{}{
+													"attr-key-5": "",
+												},
+											},
+										},
+									},
+								},
+							},
+							"attributes": map[string]interface{}{
+								"attr-key-1": "value-1",
+								"attr-key-2": "value-2",
+								"attr-key-3": "value-3",
+								"attr-key-4": "value-4",
+								"attr-key-5": "",
+								"attr-key-6": "value-6",
+							},
+							"topicId": "topic-default",
+							"userId":  "",
+						},
+						StatusCode: http.StatusOK,
+						Metadata:   types.Metadata{},
+					},
+				},
+			},
+		},
+		{
+			name: "should update timestamp field for RETL events",
+			events: []types.TransformerEvent{
+				{
+					Destination: destinationWithConfigAttributes,
+					Message: map[string]interface{}{
+						"type":    "identify",
+						"channel": "sources",
+						"traits": map[string]interface{}{
+							"timestamp": "2020-01-01T00:00:00Z",
+						},
+					},
+				},
+			},
+			want: types.Response{
+				Events: []types.TransformerResponse{
+					{
+						Output: map[string]interface{}{
+							"message": map[string]interface{}{
+								"type":      "identify",
+								"channel":   "sources",
+								"timestamp": "2020-01-01T00:00:00Z",
+								"traits": map[string]interface{}{
+									"timestamp": "2020-01-01T00:00:00Z",
+								},
+							},
+							"attributes": map[string]interface{}{},
+							"topicId":    "topic-default",
+							"userId":     "",
+						},
+						StatusCode: http.StatusOK,
+						Metadata:   types.Metadata{},
+					},
+				},
+			},
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
```

Comment:
It would be useful to add some tests that compare against the node.js implementation similar to how we done it for warehouse transformation.

https://github.com/rudderlabs/rudder-server/blob/master/warehouse/transformer/transformer_test.go#L603

We can even add fuzzing tests, that only compare that given the same input the output can be the same 

https://github.com/rudderlabs/rudder-server/blob/master/warehouse/transformer/transformer_fuzz_test.go#L534

This will give us more confidence, before we even have to test it / compare it in production.

Non-blocking comment we can add this in a follow up PR